Code/Source/RobotImporter/RobotImporterWidget.cpp

```diff
@@ -0,0 +1,114 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#include <AzCore/Utils/Utils.h>
+#include <QMessageBox>
+#include <QVBoxLayout>
+
+#include "RobotImporter/RobotImporterWidget.h"
+#include "RobotImporter/RobotImporterWidgetUtils.h"
+#include "RobotImporter/URDF/RobotImporter.h"
+
+namespace ROS2
+{
+    namespace Internal
+    {
+        AZStd::string GetBaseName(AZStd::string path)
+        {
+            QFileInfo fileInfo(path.c_str());
+            return fileInfo.baseName().toStdString().c_str();
+        }
+
+    } // namespace Internal
+
+    RobotImporterWidget::RobotImporterWidget(QWidget* parent)
+        : QWidget(parent)
+        , m_statusLabel("", this)
+        , m_selectFileButton(QObject::tr("Load"), this)
+        , m_importerUpdateTimer(this)
+        , m_robotImporter(
+              [this](RobotImporter::LogLevel level, const AZStd::string& message)
+              {
+                  switch (level)
+                  {
+                  case RobotImporter::LogLevel::Info:
+                      ReportInfo(message);
+                      break;
+                  case RobotImporter::LogLevel::Error:
+                      ReportError(message);
+                      break;
+                  }
+              })
+    {
+        setWindowTitle(QObject::tr("Robot definition file importer"));
+        QVBoxLayout* mainLayout = new QVBoxLayout(this);
+        mainLayout->setSpacing(20);
+        QLabel* captionLabel = new QLabel(QObject::tr("Select a robot definition (URDF) file to import"), this);
```

Comment:
Should this read
`Select a Unified Robot Description Format  (URDF) file to import` ?